### PR TITLE
feat: implement vendor card

### PR DESCRIPTION
**Pull request by bolbolamir**

### CODE REVIEW

File: src/base.css [hunk 1]
```diff
@@ -67,27 +67,22 @@
...
 }
 
-#root,
-#__next {
+:where(#root, #__next) {
   isolation: isolate;
 }
 
-.nonvisual {
+:where(.nonvisual) {
```
Comment (hasbeenwannabe) [inline src/base.css:85]: Put this after all the resets.

File: src/base.css [hunk 2]
```diff
@@ -96,4 +91,13 @@ body {
...
   white-space: nowrap !important;
-}
\ No newline at end of file
+}
+
+:where(ol, ul, dl, dd, dt, li) {
+  list-style: none;
+  padding: 0px;
+  margin: 0px;
+}
```
Comment (hasbeenwannabe) [inline src/base.css:100]: Missing newline.

File: src/components/button.css [hunk 3]
```diff
@@ -89,4 +89,4 @@
   block-size: var(--button-svg-size);
   min-inline-size: fit-content;
   margin-inline-start: calc(var(--button-svg-size) / 1.8);
```
Comment (hasbeenwannabe) [inline src/components/button.css]: No arbitrary calculation with CSS variables allowed.

File: src/components/vendor-card.css [hunk 4]
```diff
@@ -0,0 +1,157 @@
+:where(.vendor-card) {
+  --far-away:1000;
```
Comment (hasbeenwannabe) [inline src/components/vendor-card.css]: Unused variable. Meaningless name.

File: src/components/vendor-card.css [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+:where(.vendor-card ._name) {
+  padding-block-start: var(--length-32);
+  margin-block-end: var(--length-8);
+  text-align: center;
+  font-size: var(--length-18);
+  font-weight: 700;
+  order: 1;
+}
+
+:where(.vendor-card  ._image-wrapper) {
```
Comment (hasbeenwannabe) [inline src/components/vendor-card.css]: Never name a class `wrapper`. Try to think of a better name, e.g. `media`.

File: src/components/vendor-card.css [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+  font-weight: 700;
+}
+
+:where(.vendor-card ._rating ._rating-count) {
+  margin-block: var(--length-6);
+  margin-inline: var(--length-6);
+  color: var(--color-faded);
+}
+
+:where(.vendor-card ._rating > dl > svg:first-of-type) {
```
Comment (hasbeenwannabe) [inline src/components/vendor-card.css]: No need for `>` inside `:where()`.

File: src/components/vendor-card.css [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+  margin-block: var(--length-24);
+  background-color: var(--color-primary);
+  font-size: var(--length-12);
+  line-height: var(--length-20);
+  border-radius: var(--length-64);
+  color: var(--color-text);
+  box-shadow: var(--box-shadow-light);
+}
+
+:where(.vendor-card ._delivery > div) {
```
Comment (hasbeenwannabe) [inline src/components/vendor-card.css:147]: No need for `>` inside `:where()`. Also try not to targeting a `div`.

File: src/components/vendor-card.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
+import { VendorCard } from "./vendor-card"
+import type { ComponentProps } from "react"
```
Comment (hasbeenwannabe) [inline src/components/vendor-card.stories.tsx:1]: Import library code before your own code.

File: src/components/vendor-card.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  return (
+    <div
+      style={{
+        width: "350px",
+      }}
+    >
+      <VendorCard {...cardProps}></VendorCard>
+    </div>
+  )
+}
```
Comment (hasbeenwannabe) [inline src/components/vendor-card.stories.tsx:159]: Missing newline.

File: src/components/vendor-card.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+  },
+}
+
+export const vendorCard = () => {
+  return <VendorCard {...cardProps}></VendorCard>
+}
+
+export const vendorCardAndConstrained = () => {
+  return (
+    <div
```
Comment (hasbeenwannabe) [inline src/components/vendor-card.stories.tsx]: Make the opening `<div>` tag take a single line.

File: src/components/vendor-card.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+        width: "350px",
+      }}
+    >
+      <VendorCard {...cardProps}></VendorCard>
+    </div>
+  )
+}
+export const vendorCardAndConstrainedAndRTL = () => {
+  return (
+    <div
```
Comment (hasbeenwannabe) [inline src/components/vendor-card.stories.tsx]: Make the opening `<div>` tag take a single line.

File: src/components/vendor-card.tsx [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+
+      <dt className="nonvisual">Delivery</dt>
+      <dd>
+        <dl>
+          {delivery.price ? (
+            <>
+              <dt className="nonvisual">Method / Price:</dt>
+              <>
+                <dd className="_delivery">
+                  {<DeliveryGuyIcon role="presentation" />}
```
Comment (hasbeenwannabe) [inline src/components/vendor-card.tsx]: No need to wrap this in curly braces.

File: src/components/vendor-card.tsx [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+                  <dt className="nonvisual">Curruncy</dt>
+                </dd>
+              </>
+            </>
+          ) : (
+            <>
+              <dt className="nonvisual">Type</dt>
+              <dd className="_delivery faded">
+                <>
+                  {<TimeRemainingIcon role="presentation" />}
```
Comment (hasbeenwannabe) [inline src/components/vendor-card.tsx]: No need to wrap this in curly braces.

File: src/base.css (hunk 2 above)
Comment (bolbolamir) [reply src/base.css:100]: fixed

File: src/components/vendor-card.tsx (hunk 13 above)
Comment (bolbolamir) [reply src/components/vendor-card.tsx]: done

File: src/components/vendor-card.tsx (hunk 12 above)
Comment (bolbolamir) [reply src/components/vendor-card.tsx]: fixed

File: src/components/vendor-card.stories.tsx (hunk 11 above)
Comment (bolbolamir) [reply src/components/vendor-card.stories.tsx]: fixed

File: src/components/vendor-card.stories.tsx (hunk 8 above)
Comment (bolbolamir) [reply src/components/vendor-card.stories.tsx:1]: fixed

File: src/components/vendor-card.stories.tsx (hunk 10 above)
Comment (bolbolamir) [reply src/components/vendor-card.stories.tsx]: fixed

File: src/components/vendor-card.css (hunk 5 above)
Comment (bolbolamir) [reply src/components/vendor-card.css]: fixed

File: src/components/vendor-card.css (hunk 6 above)
Comment (bolbolamir) [reply src/components/vendor-card.css]: scoping issues

**Pull request merged**